src/viam/examples/modules/simple/main.cpp

```diff
@@ -112,21 +96,9 @@ int main(int argc, char** argv) {
             return {};
         });
 
-    Registry::register_model(mr);
-
-    // The `ModuleService_` must outlive the Server, so the declaration order
-    // here matters.
-    auto my_mod = std::make_shared<ModuleService_>(socket_addr);
-    auto server = std::make_shared<Server>();
-
-    my_mod->add_model_from_registry(server, generic, m);
-    my_mod->start(server);
-    BOOST_LOG_TRIVIAL(info) << "Module serving model " << m.to_string() << ", listening on "
-                            << socket_addr;
+    std::vector<std::shared_ptr<ModelRegistration>> mrs = {mr};
```

Comment:
Why can I not just pass {mr} below or use `auto` here? I don't know 🤷🏻 ; something something implicit conversion.

Reply:
huh, that surprises me.

Reply:
I'd expect both to work?

https://godbolt.org/z/5sPWjjjzx

What error do you get when you try?

Reply:
I get: `No matching function for call to 'make_shared' (clang  ovl_no_viable_function_in_call)`

Reply:
Hmm. Well, I don't think it is important enough to spend a lot of time on, the code you have is fine.